src/bauhaus/bauhaus.h

```diff
@@ -208,9 +208,9 @@ typedef struct dt_bauhaus_t
   int cursor_blink_counter;
 
   // colors:
-  float bg_normal;   // background without focus
-  float bg_focus;    // background with focus
-  float text;        // text color
+  GdkRGBA *bg_normal;   // background without focus
```

Comment:
Why pointers? Make it `GdkRGBA bg_normal;` and you no longer have to care about freeing it later.

Reply:
As discussed on IRC, I have removed the default colors, together with the memory allocations. Also corrected two UI elements which were using hard-coded colors: the "compress history stack" button, and the darktable label itself is now using `@fg_color` values.

For proper theme handling we need a strategy and additional work, different UI elements are following different ways to use (and sometimes not use) certain style components defined in the CSS file.